Passing [""] to data.tfe_workspace_ids.names crashes plugin

<!--
Hi there,

Thank you for opening an issue! Please note that we try to keep the this issue tracker reserved for
bug reports and feature requests related to the Terraform Cloud/Enterprise provider. If you know
your issue relates to the Terraform Cloud/Enterprise platform itself, please contact
tf-cloud@hashicorp.support. For general usage questions, please post to our community forum:
https://discuss.hashicorp.com.
-->

#### Terraform Cloud/Enterprise version

<!---
Are you using Terraform Cloud or Terraform Enterprise? If Terraform Enterprise, please include the version
Example: Terraform Enterprise v202104-1
-->
I'm using Terraform CLI to access Terraform Cloud workspaces via the data.tfe_workspace_ids 

#### Terraform version
<!---
If you are using Terraform CLI, run `terraform version` to show the version.

If you are using Terraform Cloud or Terraform Enterprise, see the Terraform version being used by a
workspace in the workspace's Overview tab.

Paste the result between the ``` marks below.
-->

```plaintext
Terraform v1.0.0
```

#### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.
-->

```terraform
variable "cloud_workspace_prefix" {
  type    = string
  default = null
}

locals {
  cloud_workspace_name = var.cloud_workspace_prefix == null ? "" : "${var.cloud_workspace_prefix}${var.environment}"
}

data "tfe_workspace_ids" "this" {
  names          = [local.cloud_workspace_name]
  organization = "my-name"
}
```

#### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->

```plaintext
Stack trace from the terraform-provider-tfe_v0.27.0_x5 plugin:
panic: interface conversion: interface {} is nil, not string
goroutine 126 [running]:
github.com/hashicorp/terraform-provider-tfe/tfe.dataSourceTFEWorkspaceIDsRead(0x68, {0xe80e00, 0xc000126c00})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-tfe/tfe/data_source_workspace_ids.go:57 +0x9d1
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xeec399, {0x10dba80, 0xc0004f1cc0}, 0xc0004897a0, {0xe80e00, 0xc000126c00})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:335 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0000a50a0, {0x10dba80, 0xc0004f1cc0}, 0xc00048d460, {0xe80e00, 0xc000126c00})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:558 +0xf7
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0004fb6e0, {0x10dba80, 0xc0004f1cc0}, 0xc00048d180)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:1105 +0x32c
github.com/hashicorp/terraform-plugin-mux.SchemaServer.ReadDataSource({0xc000558cf0, 0xc000558d20, {0xc00048da80, 0x2, 0x2}, 0xc000556300, 0x1}, {0x10dba80, 0xc0004f1cc0}, 0xc00048d180)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-mux@v0.2.0/schema_server.go:309 +0x91
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadDataSource(0xc00048dac0, {0x10dbb28, 0xc0004880c0}, 0xc00007d680)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/server/server.go:247 +0x1db
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0xeaf520, 0xc00048dac0}, {0x10dbb28, 0xc0004880c0}, 0xc000194180, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:416 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000552380, {0x10ecb80, 0xc0000fe600}, 0xc00084ea00, 0xc000558db0, 0x17150d0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1217 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc000552380, {0x10ecb80, 0xc0000fe600}, 0xc00084ea00, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1540 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:878 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:876 +0x294
Error: The terraform-provider-tfe_v0.27.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

#### Expected Behavior
<!--
What should have happened?
-->
I expected the plugin to look for names that match `""`, which would be none, and so return an empty map

#### Actual Behavior
<!--
What actually happened?
-->
The plugin crashed

#### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->
I was able to work around this by passing an empty array, instead of an empty string inside and empty array

```terraform
variable "cloud_workspace_prefix" {
  type    = string
  default = null
}

data "tfe_workspace_ids" "this" {
  names          = var.cloud_workspace_prefix == null ? [] : ["${var.cloud_workspace_prefix}${var.environment}"]
  organization = "my-name"
}
```

--- a/tfe/data_source_workspace_ids.go
+++ b/tfe/data_source_workspace_ids.go
@@ -54,6 +54,11 @@ func dataSourceTFEWorkspaceIDsRead(d *schema.ResourceData, meta interface{}) err
 	var id string
 	names := make(map[string]bool)
 	for _, name := range d.Get("names").([]interface{}) {
+		// ignore empty strings
+		if name == nil {
+			continue
+		}
+
 		id += name.(string)
 		names[name.(string)] = true
 	}

--- a/tfe/data_source_workspace_ids_test.go
+++ b/tfe/data_source_workspace_ids_test.go
@@ -222,6 +222,47 @@ func TestAccTFEWorkspaceIDsDataSource_empty(t *testing.T) {
 	})
 }
 
+func TestAccTFEWorkspaceIDsDataSource_namesEmpty(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEWorkspaceIDsDataSourceConfig_namesEmpty(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "names.#", "2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "names.0", ""),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "names.1", fmt.Sprintf("workspace-foo-%d", rInt)),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "organization", orgName),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_workspace_ids.good", "full_names.%"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "full_names.%", "1"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good",
+						fmt.Sprintf("full_names.workspace-foo-%d", rInt),
+						fmt.Sprintf("tst-terraform-%d/workspace-foo-%d", rInt, rInt),
+					),
+					resource.TestCheckResourceAttr(
+						"data.tfe_workspace_ids.good", "ids.%", "1"),
+					resource.TestCheckResourceAttrSet("data.tfe_workspace_ids.good", "ids.%"),
+					resource.TestCheckResourceAttrSet("data.tfe_workspace_ids.good", "id"),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_workspace_ids.good", fmt.Sprintf("ids.workspace-foo-%d", rInt)),
+				),
+			},
+		},
+	})
+}
+
 func testAccTFEWorkspaceIDsDataSourceConfig_basic(rInt int) string {
 	return fmt.Sprintf(`
 resource "tfe_organization" "foobar" {
@@ -356,6 +397,32 @@ resource "tfe_workspace" "bar" {
 
 data "tfe_workspace_ids" "good" {
   names        = ["workspace-foo-%d"]
+  tag_names    = ["bar"]
+  organization = tfe_workspace.foo.organization
+}`, rInt, rInt, rInt, rInt)
+}
+
+func testAccTFEWorkspaceIDsDataSourceConfig_namesEmpty(rInt int) string {
+	return fmt.Sprintf(`
+resource "tfe_organization" "foobar" {
+  name = "tst-terraform-%d"
+  email = "admin@company.com"
+}
+
+resource "tfe_workspace" "foo" {
+  name         = "workspace-foo-%d"
+  organization = tfe_organization.foobar.id
+  tag_names    = ["bar"]
+}
+
+resource "tfe_workspace" "bar" {
+  name         = "workspace-bar-%d"
+  organization = tfe_organization.foobar.id
+  tag_names    = ["bar"]
+}
+
+data "tfe_workspace_ids" "good" {
+  names        = ["", "workspace-foo-%d"]
   tag_names    = ["bar"]
   organization = tfe_workspace.foo.organization
 }`, rInt, rInt, rInt, rInt)

